source/main/cardano/CardanoNode.js

```diff
@@ -340,9 +372,9 @@ export class CardanoNode {
       _log.info('CardanoNode#stop: process is not running anymore');
       return Promise.resolve();
     }
-    _log.info('CardanoNode#stop: disconnecting from cardano-node process');
+    _log.info('CardanoNode#stop: stopping cardano-node process');
     try {
-      if (_node) _node.disconnect();
+      if (_node) _node.stop(_config.shutdownTimeout);
```

Comment:
Hopefully you should now be able to change it so that `_node` becomes the `Launcher` and is always defined and so you don't need to check.

Reply:
Indeed - we will improve this in a separate PR.